android/src/main/kotlin/vn/hunghd/flutterdownloader/TaskDbHelper.kt

```diff
@@ -12,14 +12,31 @@ class TaskDbHelper private constructor(context: Context) :
     }
 
     override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
-        if (newVersion == 4) {
-            db.execSQL("ALTER TABLE ${TaskEntry.TABLE_NAME} ADD COLUMN ${TaskEntry.COLUMN_ALLOW_CELLULAR} TINYINT DEFAULT 1")
-        } else if (oldVersion == 2 && newVersion == 3) {
-            db.execSQL("ALTER TABLE " + TaskEntry.TABLE_NAME + " ADD COLUMN " + TaskEntry.COLUMN_SAVE_IN_PUBLIC_STORAGE + " TINYINT DEFAULT 0")
-        } else {
-            db.execSQL(SQL_DELETE_ENTRIES)
-            onCreate(db)
+        update1to2(db, oldVersion)
+        update2to3(db, oldVersion)
+        update3to4(db, oldVersion)
+    }
+
+    private fun update1to2(db: SQLiteDatabase, oldVersion: Int) {
+        if (oldVersion > 1) {
+            return
+        }
+        db.execSQL(SQL_DELETE_ENTRIES)
+        onCreate(db)
```

Comment:
In the old version, this was executed when updating from 1 to 2, so I think it will work now.
https://github.com/fluttercommunity/flutter_downloader/blob/v1.6.1/android/src/main/java/vn/hunghd/flutterdownloader/TaskDbHelper.java#L59-L60